sklearn/manifold/_t_sne.py

```diff
@@ -396,7 +396,8 @@ def _gradient_descent(
         kwargs["compute_error"] = check_convergence or i == n_iter - 1
 
         error, grad = objective(p, *args, **kwargs)
-        grad_norm = linalg.norm(grad)
+        grad_norm = linalg.norm(grad,
+                                check_finite=False)
```

Comment:
I think we still need `check_finite=True`, because we can not guarantee that the output of `objective` is finite.

Note: To set `check_finite=False`, we need to make sure that our code already performed the check.